frontend/src/layouts/MainLayout.jsx

```diff
@@ -0,0 +1,15 @@
+import { Outlet } from "react-router-dom";
+import Navbar from "../components/Navbar";
+import Footer from "../components/Footer";
+
+const MainLayout = () => {
+	return (
+		<>
+			<Navbar />
+			<Outlet />
```

Comment:
Please explain what exactly the use of <Outlet/> because I'm unable to understand it and never used it before.
Thank you

Reply:
> Please explain what exactly the use of because I'm unable to understand it and never used it before. Thank you

![image](https://github.com/code-Atish/codeEditor/assets/32655186/f20f4ae7-925d-4fd0-8e5c-a7a5a84ab530)
here i created a parent route MainLayout and its children routes inside of it.
basically Outlet will be remplaced with one on the children AboutPage or ContactPage or any page wrapped inside parent route.
I used this to enhance performance to not reload Navbar and footer eachtime we navigate to a differente page.
for further explanation please check this out [https://reactrouter.com/en/main/components/outlet#outlet](url) 
[https://medium.com/@shruti.latthe/understanding-react-outlet-a-comprehensive-guide-b122b1e5e7ff](url)